clap_builder/src/builder/arg.rs

```diff
@@ -4363,8 +4363,20 @@ impl Arg {
         }
 
         debug_assert!(self.is_takes_value_set());
+
         for (n, val_name) in val_names.iter().enumerate() {
-            let arg_name = if self.is_positional() && (num_vals.min_values() == 0 || !required) {
+            let value_is_required = match self.is_positional() {
+                true => required && (num_vals.min_values() != 0),
+                false => {
```

Comment:
Sorry, I lost track of this earlier, but why a `match` on a `bool`?  I would assume `if self.is_positional` would be a more obvious

Reply:
I'll push an `if`...`else` version!

Reply:
Done.